src/resource/nonRdfData.test.ts

```diff
@@ -38,10 +38,63 @@ import {
   saveFileInContainer,
   overwriteFile,
   getFileWithAcl,
+  flattenHeaders,
 } from "./nonRdfData";
 import { Headers, Response } from "cross-fetch";
 import { WithResourceInfo } from "../interfaces";
 
+describe("flattenHeaders", () => {
+  it("returns an empty object for undefined headers", () => {
+    expect(flattenHeaders(undefined)).toEqual({});
+  });
+
+  it("returns well-formed headers as-is", () => {
+    const headers: Record<string, string> = {
+      test: "value",
+    };
+    expect(flattenHeaders(headers)).toEqual(headers);
+  });
+
+  it("transforms an incoming Headers object into a flat headers structure", () => {
+    const myHeaders = new Headers();
+    myHeaders.append("accept", "application/json");
+    myHeaders.append("Content-Type", "text/turtle");
+    const flatHeaders = flattenHeaders(myHeaders);
+    expect(flatHeaders).toEqual({
+      accept: "application/json",
+      "content-type": "text/turtle",
+    });
+  });
+
+  it("supports non-iterable headers if they provide a reasonably standard way of browsing them", () => {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const myHeaders: any = {};
+    myHeaders["forEach"] = (
+      callback: (value: string, key: string) => void
+    ): void => {
+      callback("application/json", "accept");
+      callback("text/turtle", "Content-Type");
+    };
+    const flatHeaders = flattenHeaders(myHeaders);
+    expect(Object.entries(flatHeaders)).toEqual([
+      ["accept", "application/json"],
+      ["Content-Type", "text/turtle"],
+    ]);
```

Comment:
The same change could be made here :P 

```suggestion
    expect(flatHeaders).toEqual({
      "accept", "application/json",
      "Content-Type", "text/turtle",
    });
```

Reply:
Maybe they could ^^ cd6fcff7f4e8450c8c1168087d342ca4a48a669a